File: Lab_ProblemSet/substitution.php
<?php
function main(string $key): int {
    echo 'plaintext: ';
    $word = trim(fgets(STDIN));
    $alphabet = range('a', 'z');
    $ciphertext = '';

    for ($i=0; $i < strlen($word); $i++) {
        $alphabet_index = array_search(strtolower($word[$i]), $alphabet);
        if (ctype_lower($word[$i])) {
            $ciphertext .= strtolower($key[$alphabet_index]);
            continue;
        }
        $ciphertext .= $key[$alphabet_index];
    }
    echo 'ciphertext: ', $ciphertext, PHP_EOL;
    return 0;
}


if (strlen($argv[1]) != 26 ) return 1;

main($argv[1]);
// test key:YTNSHKVEFXRBAUQZCLWDMIPGJO
?>

File: Lab_ProblemSet/substitution.py
import sys

def main(key: str) -> int:
    word = input("plaintext: ").strip()
    alphabet = [chr(i) for i in range(97, 123)]  # 'a' to 'z'
    ciphertext = ""

    for char in word:
        try:
            alphabet_index = alphabet.index(char.lower())
        except ValueError:
            continue  # Skip characters not in the alphabet

        if char.islower():
            ciphertext += key[alphabet_index].lower()
        else:
            ciphertext += key[alphabet_index]

    print(f"ciphertext: {ciphertext}")
    return 0

if __name__ == "__main__":
    if len(sys.argv[1]) != 26:
        sys.exit(1)
    main(sys.argv[1])
